libusb/os/windows_winusb.c

```diff
@@ -703,10 +705,49 @@ static void winusb_exit(struct libusb_context *ctx)
 	exit_dlls();
 }
 
+static BOOL
+get_device_io_control_result(HANDLE handle, HANDLE io_event, BOOL close, LPOVERLAPPED op)
+{
+  DWORD error = GetLastError();
```

Comment:
I don't like that the function starts with GetLastError() although I understand this is to factor out common code. The last error value is here an out-of-band parameter to this function. At least the value should be passed as a parameter, and every caller can add GetLastError() in the call. Or the DeviceIOControl() should be included in this function even if that requires a lot more parameters - it will equal DeviceIOControl's parameters plus ioevent, maybe turning it into a new DeviceIOControl_async().

Reply:
Also not sure about the "close" flag. Wouldn't it be more tidy if the caller closes the handle itself if needed, depending on the return value?

Reply:
I apologize for not being able to respond to your messages recently due to busy work. Based on your previous modification suggestions, I have re-modified the DeviceIoControl and encapsulated the asynchronous usage of DeviceIoControl into device_io_control. All the places where DeviceIoControl is used in libusb/os/windows_winusb.c have been replaced with device_io_control. The maximum waiting time for device_io_control is 2 seconds to prevent some devices from getting stuck.

I hope this modification meets your approval.
@mcuee @tormodvolden